src/world/partner/goombario.c

```diff
@@ -50,16 +153,37 @@ s32 world_goombario_can_pause(Npc* partner) {
 // has big jumptable at rodata 802BDE88
 INCLUDE_ASM(ApiStatus, "world/partner/goombario", func_802BD5D8_3174F8, Evt* script, s32 isInitialCall);
 
+#ifdef NON_EQUIVALENT //something with the symbol is broken
+ApiStatus func_802BDB30_317A50(Evt* script, s32 isInitialCall) {
+    PartnerActionStatus* partnerActionStatus = &gPartnerActionStatus;
+
+    set_time_freeze_mode(0);
+
+    if (D_802BDF64_3248B4 != 0) {
+        D_802BDF64_3248B4 = 0;
+        enable_player_input();
+    }
+
+    partnerActionStatus->actionState.b[0] = 0;
+    partnerActionStatus->actionState.b[3] = 0;
+    return ApiStatus_DONE2;
+}
+#else
 INCLUDE_ASM(ApiStatus, "world/partner/goombario", func_802BDB30_317A50, Evt* script, s32 isInitialCall);
+#endif
 
 ApiStatus func_802BDB84(Evt* script, s32 isInitialCall) {
-    s32 unk = script->owner2.npc; // todo what is this?
+    Npc* npc = script->owner2.npc;
```

Comment:
Is this not goombario? May be worth naming it after him. (applies elsewhere too)